test/quantization/test_quantize_script.py

```diff
@@ -148,91 +176,133 @@ def forward(self, x):
+
+            def forward(self, x):
+                x = self.conv(x)
+                x = self.bn(x)
+                return x
+
+        class TestModule2d(torch.nn.Module):
             def __init__(self):
-                super(TestModule, self).__init__()
-                self.sub = SubModule()
+                super(TestModule2d, self).__init__()
+                self.sub = SubModule2d()
 
             def forward(self, x):
                 x = self.sub(x)
                 return x
 
-        for tracing_mode in [True, False]:
-            eager = TestModule()
-            eager.eval()
-            if tracing_mode:
-                x = torch.rand(1, 1, 10, 10)
-                scripted_or_traced = torch.jit.trace(eager, x)
-            else:
-                scripted_or_traced = torch.jit.script(eager)
-            scripted_or_traced.eval()
+        class TestModule3d(torch.nn.Module):
+            def __init__(self):
+                super(TestModule3d, self).__init__()
+                self.sub = SubModule3d()
 
-            FileCheck().check_count("prim::CallMethod[name=\"forward\"]", 2, exactly=True) \
-                .run(str(get_forward_graph(scripted_or_traced.sub._c)))
+            def forward(self, x):
+                x = self.sub(x)
+                return x
+
+        def _test_output(eager, x):
+            for tracing_mode in [True, False]:
+                eager.eval()
+                if tracing_mode:
+                    scripted_or_traced = torch.jit.trace(eager, x)
+                else:
+                    scripted_or_traced = torch.jit.script(eager)
+                scripted_or_traced.eval()
 
-            scripted_or_traced = wrap_cpp_module(torch._C._jit_pass_fold_convbn(scripted_or_traced._c))
+                FileCheck().check_count("prim::CallMethod[name=\"forward\"]", 2, exactly=True) \
+                    .run(str(get_forward_graph(scripted_or_traced.sub._c)))
 
-            FileCheck().check_count("prim::CallMethod[name=\"forward\"]", 1, exactly=True) \
-                .run(str(get_forward_graph(scripted_or_traced.sub._c)))
+                scripted_or_traced = wrap_cpp_module(torch._C._jit_pass_fold_convbn(scripted_or_traced._c))
+
+                FileCheck().check_count("prim::CallMethod[name=\"forward\"]", 1, exactly=True) \
+                    .run(str(get_forward_graph(scripted_or_traced.sub._c)))
+
+                self.assertEqual(eager(x), scripted_or_traced(x))
 
-            x = torch.rand(1, 1, 10, 10)
-            self.assertEqual(eager(x), scripted_or_traced(x))
+        _test_output(TestModule2d(), torch.rand(1, 1, 10, 10))
+        _test_output(TestModule3d(), torch.rand(1, 1, 10, 10, 10))
 
     def test_foldbn_in_customConv2D(self):
-        # Make sure a custom Conv2D class is not folded
+        # Make sure a custom Conv2D or Conv3d class is not folded
         # as we do not know it does.
-        class CustomConv2D(torch.nn.Module):
+        class CustomConv(torch.nn.Module):
```

Comment:
wait, how is this a conv module?

Reply:
oh, nvm, this test can probably be removed, we will have more exact match for conv modules